### DISCUSSION TITLE
General Questions/suggestions/sharing-progress on Servos, Power supply, materials, layout, construction/mounting, sensors/coding...

### DISCUSSION THREAD

**Discussion by TigerKIN3** [aaronse/servo-display#3] [Q&A] [answered]
We have our concerns about power draw due to us planning to use 320 servos. We are also concerned about the wire gauge for that many servos. There are more but we haven't explored them enough.

**Comment by aaronse** [comment, marked as answer]
Hello @TigerKIN3, 320 servos, wow!   Haven't built anything that large, would love to hear from folks that have.  I know less about hardware/electrical than software.  Personally, would plan for 150ma per servo, but would expect steady state (servo not moving) will draw much less than 150ma.  So, 150ma X 320 = 48amp, I'd plan for 80% or less of supply capacity.  So 5v 60amp would be what I'd get, infact, https://www.amazon.com/gp/product/B00IWC2RLS is what I ended up buying for WTD 2.0.  Used mix of flexible silicone 14-18AWG wire (depending on expected current) for trunks going from power supply to each Pca9685 board.  

https://learn.adafruit.com/16-channel-pwm-servo-driver was very helpful to me.

**Comment by TigerKIN3** [comment]
Thank you @aaronse for the insightful information. We plan on using SG-90 servos like the ones in your video. Is there any servos problems that we should look out for in our construction.

**Comment by aaronse** [comment]
You're welcome.  Been pretty lucky with https://amzn.to/3oxd1E5, so far...  ~5% arrived broken/not-working.  Initially over ordered by 10% expecting some Duds, returning ones I didn't need...  Experienced some jittering/twitching by the servos when they're supposed to be steady, could be caused by bad/wrong wiring on my part.  

*IF* you run into jitter issues, something to consider is using the PCA9685's OE pin to cut power via your code for moments when you're not expecting the Servo's to move.  Didn't do this, feel free to share pull request if you encounter and fix this issue :-)

**Comment by TigerKIN3** [comment]
I saw you used a board where the servo motors where set on. What type of board did you use to set the servos? 
![Board for Servos](https://user-images.githubusercontent.com/91908143/137367534-472e125e-46b8-43d7-9e18-4c6d4573bb15.PNG)

**Reply by aaronse** [reply]
WTD v1.0 used "Project Foam Board", kind of thing you can buy from craft stores like Jo-Ann or Michaels (in USA).  Servos velcro'd to the board and to the tiles.  Velcro used had more give than I'd like, so...

WTD v2.0, uses 3D printed 12x 4x4 panels.  Panels are mounted onto a large metal frame.  So, only using velcro on one side and all the servos are precisely located.  Just uploaded the .stl incase you/others want pursue that path https://github.com/aaronse/ServoPixelDisplay/blob/main/models/WTD_Tile_v2.7_4x4.stl

![image](https://user-images.githubusercontent.com/16479976/137372120-f5bdc798-d171-45ac-92a3-5971933a3f01.png)

For WTD v2.0, Servos are body side up (opposite to WTD v1.0) at exact locations on the 3D printed panels.  Faster, more precise install compared to measuring and cutting foam board.  Also, mounting body side up make servo pivot point closer to the tile, enabling higher pixel density with smaller gaps between tiles.  Used these servos https://amzn.to/3oxd1E5 (affiliate link).

![image](https://github.com/aaronse/ServoPixelDisplay/blob/main/assets/wtd2.0_servos_mounted.jpg?raw=true)

**Reply by scorto** [reply]
Hello, don't you think the servo motor won't need more power in this position?
i found a fixation for the first version of the WTD https://www.thingiverse.com/thing:2904824
And perhaps it can replace your actual fixation on the WTD v2?

**Reply by aaronse** [reply]
> Hello, don't you think the servo motor won't need more power in this position? i found a fixation for the first version of the WTD https://www.thingiverse.com/thing:2904824 And perhaps it can replace your actual fixation on the WTD v2?

Yes, agree, the motors will take slightly more power (and time) to accelerate-move-decelerate the additional 9gram mass of the motor.

Thinking the servos are rated for 450g+, so hoping the 60a supply will work for the 192 servos we have.  Went with this design/layout in v2.0 to reduce time required to precisely mount all the servos to be evenly spaced.  WTD 1.0 used 2 sets of velcro, which resulted in more slop/movement than I'd like.  WTD 2.0 uses 1 set of velcro, could have used VHB tape, but want to swap out the balsa wood tiles for other material at some point.

**Reply by scorto** [reply]
I see what you mean by reducing time to precisely mount.But perhaps can be solved by making a guide in 3D print? this is what i'm thinking of. I saw Daniel Rozin having such a guide to fix panel in a video.
Look at 6:35  https://www.youtube.com/watch?v=kV8v2GKC8WA

**Reply by scorto** [reply]
<img width="358" alt="Capture d’écran 2021-11-14 à 22 31 20" src="https://user-images.githubusercontent.com/94319643/141699402-bef31f45-86d8-453a-8051-0b4005c88c73.png">
<img width="355" alt="Capture d’écran 2021-11-14 à 22 31 28" src="https://user-images.githubusercontent.com/94319643/141699410-f01ad7a0-f02c-4961-b2c9-55ebd32b3e01.png">
<img width="356" alt="Capture d’écran 2021-11-14 à 22 31 37" src="https://user-images.githubusercontent.com/94319643/141699415-87d33366-e017-44ab-84e6-b58a90f63b4c.png">

**Reply by aaronse** [reply]
Guide(s) for laying out on foamboard for v1.0 was helpful.  3D printed guide, or template made from paper/card would be great too.  Below is a pic of what I started using for v1.0.  Precise layout was a time consuming phase of v1.0 for me, the v2.0 3D printed panels were born out of frustration :-)

![16369258292037119202872241832326](https://user-images.githubusercontent.com/16479976/141699590-b727d33d-0f05-450e-82fe-6e50ddec5390.jpg)


Liked Daniel's idea of using a guide to mount the tiles as well, ended up creating one for v2.0, will share .stl if it works out.  Here's what they currently look like:

![16369258575932210081760690265584](https://user-images.githubusercontent.com/16479976/141699618-c2543f96-caa6-4c6f-a30f-61cc4a44a7fc.jpg)

**Reply by scorto** [reply]
Looks great ;) addition with this i'll at the end try to  use a raspberry with pi camera for moving the tiles. Do you?
I saw this project, and think of mixing with yours: https://github.com/julienp17/WaveMirror
Do you plan todo such feature too?

**Reply by aaronse** [reply]
Thanks for sharing WaveMirror link, looks like a neat project.  Interesting idea to 3D print cylinder like tiles ( https://github.com/julienp17/WaveMirror/blob/master/solidworks/prototypes/cylindres/Cylindre4_20181203.STL )

Totally agree on your approach, planning on Pi + Camera for v2.0 as well, Python+OpenCV.  v1.0 was C on Arduino Uno, ported to ESP8266 and ESP32.  But for camera processing will need more CPU power like Pi.  Not clear how well Pi will play with 192 servos making high frequency position updates while concurrently handling image processing.  Backup plan is to offload servo control to a ESP32.

**Reply by scorto** [reply]
I'm trying not using velcro ;)
The wavemirror uses several arduino instead of PCA9685 PWMas we do.And the treatment of the image is in row and not in square of 16 tiles likes ours. Which needs coding adjustment on the raspberry and the arduino. It is not the moment for me to see this but just having it  in mind.

**Reply by scorto** [reply]
Hi, i think of something today. Do you ever think of printer the tiles with the support to directly fix on the servo?

**Reply by aaronse** [reply]
Nice idea!  I didn't think about this, even though I'd printed a support arm with a 'tile' attached before :-) 

![image](https://user-images.githubusercontent.com/16479976/141852898-c5943a9f-1794-4022-b91e-214316e7c661.png)

What material(s) would look good and reflect light well?  Wood filament?

My v2.0 has servo body elevated, to reduce tile swing, enabling higher DPI.  Already cut 192 balsa wood tiles...  However, the 9g servos am using have notched flanges.  Maybe a tile with slightly springy clips that fit the notches would work.  Would take time to print, could be tricky to replace depending on design/print quality.  But, maybe, with some more thought this could be precise to place, and fast to install...

**Reply by scorto** [reply]
Great, i have some filament from prusa that has got very cool reflect (for exemple PLA mystic green or PLA mystic brown).
But can the tile front can be perfectly finished?

And with the tile including arm support , we can imagine for easy and precise installation, having the support for the servo i've shown you :)
 
And for easy maintenance, the tile you've made with the arm support, why not making it in 2 pieces that can be screwed?
so 3 part: Tile, arm support, servo fixation (with extended fixation arms in cross to help?). All this can be screwed together on the principal support of the project. This can save some printing if using a wood panel for the entire project...

**Reply by scorto** [reply]
Hello, i'm wondering if you can share the file of the support arm with a 'tile' attached?
What is the size of this tile? thanks

**Reply by scorto** [reply]
Hello @aaronse any news?
I've tried my new ac 5V 10A power but i don't understand, only 2 boards are powered..... and i've check all the board individually.
Any idea? 
Thank you

**Reply by aaronse** [reply]
Hello @scorto!  Images/YT of PLA mystic green and mystic brown look pretty cool!  Curious how your build turns out.  

Seen ABS vapor smoothing on YT before, but not sure how to vapor smooth PLA.  Wet sanding results can be good-to-disappointing, seen people do epoxy and lacquer sprays to smooth.  But I've never tried any of these, maybe standing back will be good enough :-)  

Another trick...  By default, first layer lines for my Ender 3 Max prints are 45degs, making for diagonal lines.  Maybe Horizontal/Vertical orientated filament lines would reflect overhead/ambient light source better though.  So, could maybe try printing tiles rotate 45deg around Z axis so that the first layer filament lines are vertical/horizontal (maybe there's a print setting for this already and Z axis rotation isn't needed)?

If 1st layer printed as-is, or with some finishing/smoothing effort, is great to look at, then...  Printing tiles with the servo arm support would be a great way to enable precise install.  Am digging around for .stl with tile and support arm, will attach when found.

"2 boards" as in 2 PCA9685 boards, with their addresses soldered appropriately?

**Reply by aaronse** [reply]
Hey @scorto, just added https://github.com/aaronse/ServoPixelDisplay/blob/main/models/WTD_Tile_v2.1_60mm.stl , am using Fusion 360.  Out of curiosity am printing rotated 45 deg on Z-axis, my goal is vertical lines on tile face.  I won't be using, but hope the model helps your project.

For my WTD 2.0 build I'd need to do something different because my servos are elevated, will add model if I end up printed the tiles.  Again, am curious to see how your color changing filaments turn out, they look cool.

**Reply by scorto** [reply]
Hello, thanks for the tile's file:) i'll print one of each of the 2 PLA color to show you the result.
For the servo, i think i found the problem:it is the cable connector of the DC that has a problem.
I'll change it and make new tries. ;)

**Reply by scorto** [reply]
@aaronse 
Here a some testing with 3 colors: PLA mystic green , mystic brown and galaxy black .

I put the attach in grey i've found on thingiverse to fix it on the support. What do you think of this?
But as you see there is a difference of max angle in each side due to the central part that is reducing the angle.

Waiting you thoughts about that
<img width="334" alt="Capture d’écran 2021-11-23 à 21 14 47" src="https://user-images.githubusercontent.com/94319643/143098390-f694ed4f-ddf3-4033-b3cb-3f7d59e56f28.png">
<img width="363" alt="Capture d’écran 2021-11-23 à 21 14 57" src="https://user-images.githubusercontent.com/94319643/143098404-2d532e76-f344-485c-b16b-0037ad277cbf.png">
<img width="366" alt="Capture d’écran 2021-11-23 à 21 15 08" src="https://user-images.githubusercontent.com/94319643/143098411-801af352-0621-4709-8d9f-2a853c279ea4.png">
<img width="348" alt="Capture d’écran 2021-11-23 à 21 15 17" src="https://user-images.githubusercontent.com/94319643/143098413-39990f33-54bb-4f30-88c1-8561ea50d541.png">
<img width="444" alt="Capture d’écran 2021-11-23 à 21 15 29" src="https://user-images.githubusercontent.com/94319643/143098414-8f711ed1-c3da-44f1-a475-bdafe871050d.png">
<img width="449" alt="Capture d’écran 2021-11-23 à 21 15 42" src="https://user-images.githubusercontent.com/94319643/143098417-26f5deed-3755-4a26-9921-bc605303bedb.png">
<img width="371" alt="Capture d’écran 2021-11-23 à 21 15 58" src="https://user-images.githubusercontent.com/94319643/143098421-637be53b-4010-41fc-9f1a-60c87aeb9c29.png">
<img width="312" alt="Capture d’écran 2021-11-23 à 21 16 13" src="https://user-images.githubusercontent.com/94319643/143098423-b787e848-cb16-4b13-b429-8ffcac428fe8.png">
<img width="331" alt="Capture d’écran 2021-11-23 à 21 16 28" src="https://user-images.githubusercontent.com/94319643/143098424-2ef72cf7-5c0a-4a99-b2fd-7f6b2dc49778.png">
<img width="417" alt="Capture d’écran 2021-11-23 à 21 16 40" src="https://user-images.githubusercontent.com/94319643/143098425-bbc89fb4-062f-4e6a-b732-42e1bf245b8f.png">
<img width="401" alt="Capture d’écran 2021-11-23 à 21 16 51" src="https://user-images.githubusercontent.com/94319643/143098428-6c5928e6-eb6c-4111-8483-8265d9192914.png">
<img width="438" alt="Capture d’écran 2021-11-23 à 21 17 02" src="https://user-images.githubusercontent.com/94319643/143098429-784e33e8-05a6-491b-9edf-fdcb07a0d2cb.png">
<img width="465" alt="Capture d’écran 2021-11-23 à 21 17 16" src="https://user-images.githubusercontent.com/94319643/143098431-e4af4a3a-0825-4017-9b9f-aff3c20a2741.png">
<img width="708" alt="Capture d’écran 2021-11-23 à 21 17 29" src="https://user-images.githubusercontent.com/94319643/143098434-91170c8c-1a5f-4564-9fcd-d4eae88be09d.png">
<img width="558" alt="Capture d’écran 2021-11-23 à 21 17 41" src="https://user-images.githubusercontent.com/94319643/143098439-beeffc29-a6b8-4032-bfc8-5427268dccaf.png">
<img width="779" alt="Capture d’écran 2021-11-23 à 21 17 56" src="https://user-images.githubusercontent.com/94319643/143098440-24760ded-70f6-411e-907b-1a61f2c03637.png">

**Comment by scorto** [comment]
Hello @aaronse new support found for the PCA9685
Very well done and allows to organised the back ;)
https://www.thingiverse.com/thing:3430275
<img width="293" alt="Capture d’écran 2021-11-28 à 11 34 22" src="https://user-images.githubusercontent.com/94319643/143764673-a9c27e5a-4862-470f-b782-e684cf3fe5c9.png">

**Reply by aaronse** [reply]
Nice PCA9685 holder.  Sounds like great idea when using Wood/MDF/solid backing to mount everything.  For my WTD 2.0 have been mounting PCA9685 to back of the PLA panels that support the servos.  WTD 2.0 backside is less messy than 1.0 rats nest, but still more ugly than I'd like.

**Comment by scorto** [comment]
Hello  @aaronse 
Still finding new interesting things for the tile colors.
I've just discovered Dual-Color PLA ,looks amazing.
Will order some to make test.
Seethe US link: https://www.matterhackers.com/store/c/matterhackers-quantum-pla

**Reply by aaronse** [reply]
@scorto how do the Galaxy Black tiles look IRL, do you like?  Waiting for Galaxy Black PETG to be in-stock on Prusa again, AMZN prices are crazy.

Agree, the Dual color filaments look amazing.  Recently stumbled onto dual color info/overview CNC Kitchen provides at https://www.youtube.com/watch?v=Y5EC_qRTojE  Will be interesting to see how filament can be placed with consistent dual color orientation for all the tiles, curious how your tiles turn out.

Ended up designing 2" tiles with (fragile...) clips for mounting directly to servo body.  The tiles can be precisely placed and have less arm swing than 1.0, enabling smaller gaps between tiles.  Experimented with different PLA/Wood materials and filament orientations.  PETG would flex better during install/release.

![image](https://user-images.githubusercontent.com/16479976/143788218-c59d398b-0496-4043-8a1a-9a8088c5e897.png)

Tile install is quick, but unfortunately placement onto the servos is too precise...  Servo quality varies, expected given the price, many of the mounted servos have too much play and are not perfectly aligned, this results in tiles being noticeably tilted 🤦‍♂️ I should have designed the panels with smaller gap between the servo arm mount and servo body (0.25mm to 0.5mm).  

![image](https://user-images.githubusercontent.com/16479976/143787904-ec75bc69-2232-4f9c-929d-5d659bfd54ff.png)

All 192 servos are already mounted, argh...  For my build, I might try 3D printing slide-on washers to insert for tilted servos.  Another option is to ditch printed tiles and use wood+velcro.  Wood tiles seem to be easier to place level when using a 3d printed guide tool.  That said, simple matte white tiles looked better than expected IRL.

Still tinkering...

**Reply by scorto** [reply]
Hello @aaronse 
Agree, some servos are not well aligned. I saw a video that shows how to find min and max travel but huge work for 192 servos..... you have to test each one. 
From my test, velcro is not precise. I fixed each servo on the board with the support i've posted. And it is better.
I  think as the servos are not either precise and resistant, it is reasons why Daniel Rozin use stepper motors.....
Is it possible for you to change a little the tile file with deleted the central support that doesn't allow the servo to fully rotate, or to elevate the support of the arm of the servo? This to allow better movement? Don't you have this problem?

**Reply by scorto** [reply]
@aaronse here is the video for centering each servo
https://www.youtube.com/watch?v=JaVC_ZA2C-s

**Reply by aaronse** [reply]
Thanks for the centering video.  Have been rotating each servo to 0 position by hand before mounting at ~30 deg tilt, coding tilt range -30deg to 30deg, expecting to micro adjust each tile thru SW offset post install.  Automated calibration would be cool, will see...

Agree about steppers being better quality.  For me, they were bit too expensive, and the wiring, power, control seemed to be more complex.

For my build, the WTD 2.0 mounts are intentionally designed for the servo mount flanges to hit the base.  In my case the tiles will probably be velcro'd, and I don't want the servos moving so far the tiles are removed.  Am relying on the servo mount flanges to provide a hard stop preventing tile from ever touching the base.  Note this is a missed accoustic opportunity though, Daniel Rozin's metal tile and metal base builds make some interesting sounds (music?) when the tiles are intentionally smashed into the base :-)

For your build, sounds like some tile dimension changes would help, let me know precise change need, will try to make.

**Reply by scorto** [reply]
Hello @aaronse i've just received bi color pla and will make some try and pos photos ;)
for the tile model, would you accept making an hexagone shape by extending the side?
i would like to make some to  seethe effect with this kind of shape.
I made a sketch as i could.....
thank
<img width="374" alt="Capture d’écran 2021-12-04 à 10 28 53" src="https://user-images.githubusercontent.com/94319643/144704684-852389d6-d112-4ccf-99e6-0185876f539c.png">
s

**Reply by aaronse** [reply]
Hexagon will be interesting, what dimensions for hex tile?

**Reply by scorto** [reply]
I was thinking of the same dimension for bottom and top line 6cm and so extension related to this for the other sides.
What do you think of that?
I'm not sure perhaps the actual 6cm tile has to enter the hex to get enough place to hide the servo?

**Reply by scorto** [reply]
Result of the bicolor pla testing
<img width="438" alt="Capture d’écran 2021-12-04 à 12 32 24" src="https://user-images.githubusercontent.com/94319643/144707963-52e0a41c-5639-498e-9fec-df8223d168ed.png">
<img width="415" alt="Capture d’écran 2021-12-04 à 12 32 37" src="https://user-images.githubusercontent.com/94319643/144707966-f55be8b1-3168-4a82-8d8b-ac46da47d8ed.png">
<img width="362" alt="Capture d’écran 2021-12-04 à 12 32 47" src="https://user-images.githubusercontent.com/94319643/144707968-6e1cc83f-1683-4bbc-91fb-2de3e5bd00a6.png">
<img width="394" alt="Capture d’écran 2021-12-04 à 12 32 59" src="https://user-images.githubusercontent.com/94319643/144707969-2a89cd79-3a5e-4efc-b2fa-5d916378304d.png">
<img width="295" alt="Capture d’écran 2021-12-04 à 12 33 06" src="https://user-images.githubusercontent.com/94319643/144707970-d72d4e2c-3e27-4e35-8d01-631447d6decc.png">
<img width="336" alt="Capture d’écran 2021-12-04 à 12 33 14" src="https://user-images.githubusercontent.com/94319643/144707971-07c74a8d-989d-435c-ad15-56ca3f25e8d7.png">
<img width="259" alt="Capture d’écran 2021-12-04 à 12 33 20" src="https://user-images.githubusercontent.com/94319643/144707973-943781b5-b503-40eb-af37-bbed6ee74795.png">
<img width="387" alt="Capture d’écran 2021-12-04 à 12 33 35" src="https://user-images.githubusercontent.com/94319643/144707974-83ec9be6-d950-429f-96f3-872a7799eefb.png">

**Reply by aaronse** [reply]
Hexagon with 60mm sides might be large enough to hide, will see later today...

**Reply by scorto** [reply]
@aaronse i've tried to make the file. Not sure it is perfect?
Now my problem is to perfect align the tiles with the same spaces, not very easy with the type of support i use.....
<img width="156" alt="Capture d’écran 2021-12-05 à 22 17 11" src="https://user-images.githubusercontent.com/94319643/144764264-076dadca-999e-401d-871e-21e3622fe054.png">

<img width="456" alt="Capture d’écran 2021-12-05 à 22 12 21" src="https://user-images.githubusercontent.com/94319643/144764228-9579393d-1097-46a2-bc23-c788f9236c43.png">
<img width="443" alt="Capture d’écran 2021-12-05 à 22 12 40" src="https://user-images.githubusercontent.com/94319643/144764230-34d11bc8-91c3-4962-a636-339ad3e911e6.png">
<img width="464" alt="Capture d’écran 2021-12-05 à 22 13 05" src="https://user-images.githubusercontent.com/94319643/144764232-4c23089a-c323-481a-92f4-cbe26dd591c2.png">

**Reply by aaronse** [reply]
Great job on the hex tiles!  Curious if you're finding hex tiles need larger or smaller gap between them (than square tiles) to avoid collisions during motion?  Currently planning on 3/8" gap between WTD 2.0 square tiles.

Using crafty thoughtful coding, guessing you could get away with very small gaps via some algorithm (e.g. constrain tiles to rotate in the same or less angle than the neighbor tile(s) being rotated towards).  Could add more conditional constraining logic to allow tiles to move freely but adhere to collision avoidance algorithm when rotating at extreme enough angles for collision potential.  Writing for hex layout will be interesting, since neighbor count is more.  Personally, I'd be very tempted to leave large enough gap so collision isn't physically possible, enabling my code to be simpler :-)

**Comment by scorto** [comment]
@aaronse look at that :) 
https://www.instagram.com/reel/CVi3zqGDQNj/

https://www.youtube.com/watch?v=jHo4l41cz8s&t=522s

Amazing no???

**Reply by aaronse** [reply]
Totally agree, great find @scorto!  YouTube video has lots of great info about how they designed, built and coded.

**Reply by scorto** [reply]
Yes, plenty of interessting information and the mecanism with 3 motors to allow having the tile very closed :)

**Reply by scorto** [reply]
Hello @aaronse here is a new test on my side.
<img width="382" alt="Capture d’écran 2021-12-10 à 20 35 33" src="https://user-images.githubusercontent.com/94319643/145631779-465b3842-553b-4db3-be4d-ef650a12e2f3.png">
<img width="801" alt="Capture d’écran 2021-12-10 à 20 35 42" src="https://user-images.githubusercontent.com/94319643/145631794-6e9c5900-b5f4-4287-807f-610a5f4391dd.png">
<img width="457" alt="Capture d’écran 2021-12-10 à 20 35 59" src="https://user-images.githubusercontent.com/94319643/145631803-de91293c-d068-4456-b3c1-e3572bdfc0ba.png">
<img width="460" alt="Capture d’écran 2021-12-10 à 20 36 09" src="https://user-images.githubusercontent.com/94319643/145631805-41fe3ed5-ca59-4a94-bc22-cce5a99313b3.png">
<img width="466" alt="Capture d’écran 2021-12-10 à 20 36 20" src="https://user-images.githubusercontent.com/94319643/145631810-edc00669-b793-4547-aa9e-8e790cc696a8.png">
<img width="425" alt="Capture d’écran 2021-12-10 à 20 36 31" src="https://user-images.githubusercontent.com/94319643/145631817-c20d9374-8c05-4e14-98d6-0b03f160d0cf.png">

and you what's new? any new photos?

**Reply by aaronse** [reply]
Nice uniform tile layout and assembly!

**Comment by scorto** [comment]
Hello @aaronse 
What's new?
did you ever seen this?  https://www.youtube.com/watch?v=jHo4l41cz8s&t=82s

**Reply by aaronse** [reply]
Hey @scorto!  Yep, very cool project, detailed video, thanks again for sharing.

- 1:48, 2:02, 2:20, 3:20, 4:55, back and side views of Glass mirror tile servo mounts and support.
  - 2 axis servos, ball joint assembly, printer friendly pivots, plywood backing
- 2:25 Q: Curious why machine screws in tile mount behind glass?  Maybe glass was machined with recessed holes to help with precise aligning mounting to the tile mount.
- 3:39 PCA9865 16 channel servo drivers .
- 3:55 PI 3, Mean Well LRS-35-5 Power Supply.
- 4:20 Bulky, but interesting looking clear acrylic wireless remote control for interacting and mirror calibration...
  - 2 thumbs, slider, motion sensor.
- 5:09 Mirrors from MNG Ecological.
- 6:18 Rhinoceros 6  Grasshopper 3D used for creating animation sequences.

No updates/photos from me, holiday season, have put WTD 2.0 on hold to finish up a lighting project.

**Reply by scorto** [reply]
Here is another version by the same maker, with 3 arm for each tile
https://www.instagram.com/reel/CWtJ1DcAcze/?utm_source=ig_web_copy_link

**Reply by aaronse** [reply]
Blimey!  Writing the choreography software for that dance must have been head scratching fun.

**Reply by scorto** [reply]
Yes awesome :)

**Reply by scorto** [reply]
Hello @aaronse what's new about the project?

**Reply by aaronse** [reply]
> Hello @aaronse what's new about the project?

Hello @scorto, happy new year!  

Servos are wired and powered.  Hooking up and remote coding Raspberry PI Zero 2 (Python), using another PI Cam to remote view while coding since WTD 2.0 is handing on the wall now.  Might end up switching to C++ on ESP32 depending on refresh rate.  If end up using ESP32, then will be using esp8266-iot-framework compiled to ESP32 (something am currently porting).

**Reply by scorto** [reply]
Hello happy new year too ;)
sounds great all that, any photos?
